### ISSUE TITLE
imagetagger

### ISSUE THREAD

**Issue by terial** [dronekit/dronekit-la#109]
I see there is the option to make imagetagger in the make file. I have never seen anything about this or gotten it to successfully build. Anybody have any input on what it does or documentation?

**Comment by terial** [comment]
YEAH! if you could get it working I would happily use it for my past data and future. I fly roughly once a week for our aerial topos. Let me know if there is anything I can do.

**Comment by peterbarker** [comment]
On Mon, 20 Nov 2017, terial wrote:

> 
> YEAH! if you could get it working I would happily use it for my past data
> and future. I fly roughly once a week for our aerial topos. Let me know if
> there is anything I can do.

OK, I've pushed up something in a very-much "compiles for me" state.

I've lost track of whatever data sets I had for testing this code, 
however!

If you could supply something I could test against that might make the 
round-trip time to getting this working smaller.

Yours,
-- 
Peter Barker                          |   Programmer,Sysadmin,Geek.
pbarker@barker.dropbear.id.au	      |   You need a bigger hammer.
:: It's a hack! Expect underscores! - Nigel Williams

**Comment by terial** [comment]
Thanks Peter, running make right now. Is there anything I should be aware of when running it? I assume it is similar to what mission planner does to tag images (direct it to a log and also a directory containing the images)? I also assume that it only works with .JPG and not TIFF? Is it looking for the TRIG or CAM messages in the .bin?

**Comment by peterbarker** [comment]
On Thu, 23 Nov 2017, terial wrote:

> Thanks Peter, running make right now. Is there anything I should be aware of
> when running it? I assume it is similar to what mission planner does to tag
> images (direct it to a log and also a directory containing the images)? I

I've now had a closer look.  Remember this was a POC :-)

It requires a tlog, and uses the gps_raw_int (not global_position_int, 
which is odd) and mount status messages to take images based on the first 
tlog message it receives after the date on the image.

> also assume that it only works with .JPG and not TIFF? Is it looking for the

Whatever exiftool handles - it just shells out to get the data.

That being aid, it currently seems to be hard-coded to skip anything other 
than JPG.

It's been over 2 years since I looked at this tool :-)
